### DISCUSSION TITLE
DialogTrigger: usage with functional component

### DISCUSSION THREAD

**Discussion by AdisonCavani** [radix-ui/primitives#2228] [Help] [answered]
I'd like to separate my `DialogTrigger` (child) to a separate component (to separate large business logic).

```tsx
import { Dialog, DialogContent, DialogTrigger } from "@radix-ui/react-dialog";
import { forwardRef } from "react"

function Test() {
  return (
    <Dialog>
      <DialogTrigger asChild>
        <Button/>
      </DialogTrigger>
      <DialogContent>
        Witam graczy!
      </DialogContent>
    </Dialog>
  )
}

export default Test

// This would be in a separate file
const Button = forwardRef<HTMLButtonElement>((_, ref) => {
  return <button ref={ref}>Click me</button>
});

Button.displayName = "Button"
```

When I click on the `Button` component, the dialog is not opened.
The dev console is clear - no errors reported.

**Comment by joaom00** [comment, marked as answer]
You should forward all props too https://www.radix-ui.com/docs/primitives/guides/composition#composing-with-your-own-react-components

**Reply by AdisonCavani** [reply]
How to compose component in Typescript?
The docs are missing which types to use. It cannot be any

**Reply by joaom00** [reply]
I think this section should be on the composition page :thinking:

https://www.radix-ui.com/docs/primitives/guides/styling#extending-a-primitive
